huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go

```diff
@@ -0,0 +1,535 @@
+		Password:             domain["password"].(string),
+		ActiveDomainIp:       domain["active_domain_ip"].(string),
+		AcitveDomainName:     domain["active_domain_name"].(string),
+		StandyDomainIp:       domain["standby_domain_ip"].(string),
+		StandyDomainName:     domain["standby_domain_name"].(string),
+		ActiveDnsIp:          domain["active_dns_ip"].(string),
+		StandyDnsIp:          domain["standby_dns_ip"].(string),
+		DeleteComputerObject: utils.Int(isDeleteObject(domain["delete_computer_object"].(bool))),
+	}
+
+	return &result
+}
+
+func buildServiceNetworkIds(networkIds []interface{}) []services.Subnet {
+	if len(networkIds) < 1 {
+		return nil
+	}
+
+	result := make([]services.Subnet, len(networkIds))
+	for i, networkId := range networkIds {
+		result[i] = services.Subnet{
+			NetworkId: networkId.(string),
+		}
+	}
+
+	return result
+}
+
+func buildServiceCreateOpts(d *schema.ResourceData) services.CreateOpts {
+	return services.CreateOpts{
+		AdDomain:             builServiceAuthConfiguration(d),
+		VpcId:                d.Get("vpc_id").(string),
+		Subnets:              buildServiceNetworkIds(d.Get("network_ids").([]interface{})),
+		AccessMode:           d.Get("access_mode").(string),
+		EnterpriseId:         d.Get("enterprise_id").(string),
+		DedicatedSubnets:     strings.Join(utils.ExpandToStringList(d.Get("dedicated_subnets").([]interface{})), ";"),
+		ManagementSubnetCidr: d.Get("management_subnet_cidr").(string),
+	}
+}
+
+func refreshServiceStatusFunc(client *golangsdk.ServiceClient) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := services.Get(client)
+		if err != nil {
+			return resp, "", err
+		}
+
+		return resp, resp.Status, nil
+	}
+}
+
+func waitForServiceCreateCompleted(ctx context.Context, client *golangsdk.ServiceClient, timeout time.Duration) (string,
+	error) {
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"SUBSCRIBING"},
+		Target:       []string{"SUBSCRIBED"},
+		Refresh:      refreshServiceStatusFunc(client),
+		Timeout:      timeout,
+		Delay:        10 * time.Second,
+		PollInterval: 15 * time.Second,
+	}
+
+	resp, err := stateConf.WaitForStateContext(ctx)
+	return resp.(*services.Service).ID, err
+}
+
+func resourceServiceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	client, err := conf.WorkspaceV2Client(conf.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating Workspace v2 client: %s", err)
+	}
+
+	createOpts := buildServiceCreateOpts(d)
+	_, err = services.Create(client, createOpts)
+	if err != nil {
+		return diag.Errorf("error creating Workspace service: %s", err)
+	}
+	enterpriseId, err := waitForServiceCreateCompleted(ctx, client, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.Errorf("an error occurred while subscribing the service: %s", err)
+	}
+
+	d.SetId(enterpriseId)
```

Comment:
enterpriseId is a field of schema, and can been updated, pls rename the var: `enterpriseId`

Reply:
done. the name of this parameter is `resourceId` now.